utils/generator-utils.stanza

```diff
@@ -221,6 +209,38 @@ public defn set-power-source (pos:JITXObject, neg:JITXObject, v-in:Double):
     property(pos.gnd-ref) = neg
     property(pos.power-supply-pin) = PowerSupplyPin(v-in, 0.3) 
 
+public defn propagate-net-voltages () :
+  inside pcb-module:
+    for item in all-connected-items(self) do :
+      
+      ;First gather all the voltages on all connected nets.
+      val voltages = to-tuple $ seq?({property?(_.net-voltage)}, nets(item))
+      ; val voltages = Vector<Toleranced>()
+      ; for n in nets(item) do :
+      ;   if has-property?(n.net-voltage) :      
+      ;     add(voltages, property(n.net-voltage))
+      ;If at least one net has a voltage property.
+      if not empty?(voltages) :
+
+        ;Gather all the net names (some nets don't have names.).
+        val net-names = Vector<String|Ref>()
+        for (net in nets(item), c in 0 to false) do :
+          match(ref?(net)) :
+            (f:False) : add(net-names, to-string("unnamed-net-%_" % [c]))
+            (r:Ref) : add(net-names, r)
+
+        ;Let's check that all the voltages happen to be the same.
+        ;If they are different, then something has gone wrong.
+        if length(voltages) > 1 :
+          for i in 1 to length(nets(item)) do :
+            if voltages[i] != voltages[0] :
+              fatal("Mismatched net-voltage on net(s) %," % [net-names])
+
+        ;Now assign the voltage to all of the component pins in 
+        ;the module.
+        for p in component-pins(item) do :
+          property(p.net-voltage) = voltages[0]
```

Comment:
The original code was
```
if not empty?(voltages) : 
  for p in component-pins(item) do :
  property(p.net-voltage) = voltages[0]
```
The new version will crash on `voltages[0]` when voltages is empty, which is very often the case.

Reply:
This version is good:
```
        if not empty?(voltages) :
          if length(unique(voltages)) != 1 : 
            fatal("Mismatched voltages %_ on nets %_" % [voltages, net-names])
          ;Now assign the voltage to all of the component pins in the module.
          for p in component-pins(item) do :
            property(p.net-voltage) = voltages[0]

Reply:
Ok I just saw that you had put a `if not empty?(voltages) :` at the very beginning, I missed it